### PR TITLE
Add proxy port to CNP if needed

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/29873

## Checklist

I have:

- [x] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/resource/ciliumnetpol/resource.go [hunk 1]
```diff
@@ -46,6 +48,43 @@ func toCiliumNetworkPolicy(v interface{}) (*unstructured.Unstructured, error) {
...
+		}
+	}
+	if ok {
+		proxyURL, err := url.Parse(proxyString)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		proxyPort := proxyURL.Port()
+		if proxyPort == "" {
+			proxyPort = "80"
```
Comment (QuentinBisson) [inline service/controller/resource/ciliumnetpol/resource.go:83]: Well this is nasty but if it works

**Review by QuentinBisson — APPROVED** [review]

**Review by QuantumEnigmaa — APPROVED** [review]
I like nasty code :)

**Comment by marieroque** [comment]
Need to be tested on 

- [ ] `goat`
- [ ] CAPI MC
- [ ] Vintage MC

File: service/controller/resource/ciliumnetpol/resource.go (hunk 1 above)
Comment (marieroque) [reply service/controller/resource/ciliumnetpol/resource.go:83]: Do you have a suggestion to avoid that ?

Comment (QuentinBisson) [reply service/controller/resource/ciliumnetpol/resource.go:83]: Well, the think is if it is https, it should be 443 and not 80 but no the nasty part was more about the proxy things above btw

Comment (marieroque) [reply service/controller/resource/ciliumnetpol/resource.go]: Code refactored, tell me ;-)

**Review by QuentinBisson — APPROVED** [review]

File: service/controller/resource/ciliumnetpol/resource.go (hunk 1 above)
Comment (QuentinBisson) [reply service/controller/resource/ciliumnetpol/resource.go]: Nice :)

**Pull request merged**